api/graphql/types/recurring_reservation/serializers.py

```diff
@@ -126,3 +134,241 @@ class Meta(RecurringReservationCreateSerializer.Meta):
+                description=instance.description,
+                begin=period["begin"],
+                end=period["end"],
+                user=reservation_user,
+                type=reservation_type,
+                recurring_reservation=instance,
+                age_group=instance.age_group,
+                buffer_time_before=instance.reservation_unit.buffer_time_before,
+                buffer_time_after=instance.reservation_unit.buffer_time_after,
+            )
+            through = ThroughModel(
+                reservation=reservation,
+                reservationunit=instance.reservation_unit,
+            )
+            reservations.append(reservation)
+            through_models.append(through)
+
+        Reservation.objects.bulk_create(reservations)
+        ThroughModel.objects.bulk_create(through_models)
+
+    def get_reserved_timespans(
+        self,
+        start_date: datetime.date,
+        end_date: datetime.date,
+        reservation_unit: ReservationUnit,
+    ) -> list[TimeSpanElement]:
+        time_spans: list[TimeSpanElement] = []
+        current_period: ReservationPeriod | None = None
+
+        periods: Iterable[ReservationPeriod] = (
+            Reservation.objects.overlapping_period(
+                period_start=start_date,
+                period_end=end_date,
+            )
+            .affecting_reservations(
+                reservation_units=[reservation_unit.pk],
+            )
+            .values("begin", "end")
+            .order_by("begin")
+        )
+
+        # Go through all affected reservations' periods and merge overlapping ones
+        for period in periods:
+            if current_period is None:
+                current_period = period
+                continue
+
+            # If periods overlap or touch, merge them
+            if period["begin"] <= current_period["end"]:
+                current_period["end"] = period["end"]
+                continue
+
+            time_spans.append(
+                TimeSpanElement(
+                    start_datetime=current_period["begin"],
+                    end_datetime=current_period["end"],
+                    is_reservable=False,
+                )
+            )
+            current_period = period
+
+        # Add the remaining period after merging
+        if current_period is not None:
+            time_spans.append(
+                TimeSpanElement(
+                    start_datetime=current_period["begin"],
+                    end_datetime=current_period["end"],
+                    is_reservable=False,
+                )
+            )
+
+        return time_spans
+
+    def get_reservable_timespans(self, instance: RecurringReservation) -> list[TimeSpanElement]:
+        begin_time: datetime.time = instance.begin_time  # type: ignore[assignment]
+        end_time: datetime.time = instance.end_time  # type: ignore[assignment]
+        resource = instance.reservation_unit.origin_hauki_resource
```

Comment:
`resource` -> `hauki_resource`
(to not confuse with `resources.models.Resource`)